File: leetcode/easy/Array/1_Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer, Integer> hm = new HashMap<>();
        int[] answer = new int[2];
        answer[0] = 0;
        answer[1] = 0;
        
        for(int i = 0; i < nums.length; i++){
            
            if(hm.containsKey(target-nums[i])){
                answer[0] = hm.get(target-nums[i]);
                answer[1] = i;
                return answer;
            }
            else{
                hm.put(nums[i], i);
            }
        }
        return answer;
    }
}

File: leetcode/easy/HashTable/cpp/1_Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
    
        vector<int> ans;
        unordered_map<int, int> mp;
        for(int i = 0; i < nums.size(); i++)
        {
            if(mp.find(target - nums[i]) != mp.end())
            {
                ans.emplace_back(mp[target - nums[i]]);
                ans.emplace_back(i);
            }
            else
                mp[nums[i]] = i;
        }
        
       return ans;
    }     
};